curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupMember.java

```diff
@@ -121,8 +128,11 @@ public void close()
     {
         ImmutableMap.Builder<String, byte[]> builder = ImmutableMap.builder();
         boolean thisIdAdded = false;
-        for ( ChildData data : cache.getCurrentData() )
+
+        Iterator<ChildData> iterator = cache.stream().filter(parentPathFilter(membershipPath)).iterator();
```

Comment:
Not sure if it is better, just want to mention you could stay with the more pleasant for each syntax by relying on `Iterable` like that:
Iterable<ChildData> iterable = getChildDataIterable();
        for (ChildData data: iterable)
While `getChildDataIterable` is:
```
private Iterable<ChildData> getChildDataIterable() {
        Stream<ChildData> onlyMembers = cache.stream().filter(parentPathFilter(membershipPath));
        return onlyMembers::iterator;
    }
```